design/vision.md

```diff
@@ -6,7 +6,16 @@ Github PR.
 [neovim]: https://neovim.io/charter/
 
 # Project Goals
+* Suggestion: Provide a powerful manager for a single user's dotfiles.
+* Suggestion: Modularise configuration and setup scripting.
+* Suggestion: Allow the users to share their dotfiles as packages.
+* Suggestion: Enable users to harness packages from their peers.
 
 # Project Non-Goals
+* Suggestion: Become a package manager.
+* Suggestion: Manage plugins of any other applications.
 
 # Guiding Principles
+* Suggestion: Stay in scope.
+* Suggestion: Easy does it: usability over complexity.
+* Suggestion: Contribution-friendly: low barrier to entry, easy to understand codebase.
```

Comment:
Since this is gonna be written in Rust by beginners, I'd love it if it was readable for other beginners. So it's basically a kinda advanced Rust for beginners thing.